src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,321 @@
+import React, { useContext } from 'react'
+import { add, addDays, addMonths, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '..'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./TimeSeriesChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import { ceil } from 'lodash'
+import addHours from 'date-fns/addHours'
+
+export interface TimeSeriesChartProps {
+    title?: string
+    intervalType?: "Week" | "Month" | "SixMonth" | "Day",
+    intervalStart: Date,
+    data: any[] | undefined,
+    dataGap?: Duration,
```

Comment:
"dataGap" is definitely of a weird name; as much as "interval" is overloaded it probably makes more sense to call this "interval" since that is what recharts axes would call this I would think;  or perhaps "betweenDataInterval" or something...

Reply:
> "dataGap" is definitely of a weird name; as much as "interval" is overloaded it probably makes more sense to call this "interval" since that is what recharts axes would call this I would think; or perhaps "betweenDataInterval" or something...

Yes - I wasn't totally happy with this name.  I didn't think `interval` was specific enough.  The reason that I didn't call it something like `betweenDataInterval`, `expectedDataInterval`, or `expectedDataFrequency` was that I was hoping to convey that this variable was what was helping to display *gaps* in the data.  That is, I was trying to indicate why you would or would not bother to set the value.  I probably like `expectedDataInterval` the most at this point?

Reply:
Ah yeah, I like expectedDataInterval or expectedDataFrequency

"Gap" feels weird because it's a negative; the idea of "gaps" is kind of an internal library thing